### DISCUSSION TITLE
Feature request: Allow shippingMode and shippingSolution filters in generateTransportationOptions in FBA Inbound API v2024-03-20

### DISCUSSION THREAD

**Discussion by rosh3000** [amzn/selling-partner-api-models#4426] [Ideas]
Problem: Too many options (100+) are provided when using: **ListTransportationOptions**.
I'm specifically looking for an option with **shippingMode**: _GROUND_SMALL_PARCEL_  and **shippingSoultion**: _AMAZON_PARTNERED_CARRIER_ . 99% of options returned are not suitable* and it takes several calls to cycle through (using tokens and the max page size 20) to find even one suitable option.
 
(My program currently stops when the first suitable option is found but this means it could be missing any new suitable transportation options)

Possible solution: Allow **shippingMode** and **shippingSoultion** filters in **generateTransportationOptions** operation in generateTransportationRequest->shipmentTransportationConfigurations.

This will greatly reduce the number of calls needed and show me if new options which are suitable are available

*the majority of options are for **shippingMode**: _OCEAN_LCL_, _OCEAN_FCL_, _AIR_SMALL_PARCEL_, _AIR_SMALL_PARCEL_EXPRESS_... and **shippingSolution**:  _USE_YOUR_OWN_CARRIER_

thank you

**Comment by MetaMan2023** [comment]
100% agree -- there are over 100 options for strange named carriers I would never use.   Some kind of filter system -- and if there is going to be a filter system it would be great for it to be flexible not only APC, GROUND, LTL but maybe premium carriers or specific carrier names or maybe carriers that are < a certain cost.

Some of the carriers returned are not even in English -- the carrier name is in strange unreadable characters.

Does anyone understand the "USE_YOUR_OWN_CARRIER" tag -- it seems confusing -- if I am suppose to use my own carrier, why is there a carrier listed?

**Reply by rosh3000** [reply]
Yea many are from China, USA, Germany which don't opertate in the warehouse destination (UK) for the transportationOption

I think only certain carrier operate at Amazon warhouses beacuse they have to book there delivery slot.

USE_YOUR_OWN_CARRIER are somtimes cheaper (we use to use ParcelForce) and odd occations the AMAZON_PARTNERED_CARRIER price is ridiculously high (but drops if you generateShipmentOptions a few days later).

**Reply by MetaMan2023** [reply]
FYI I have a thread to track everyone's status on the new API transition and issues -- check it out if you get a chance.

https://github.com/amzn/selling-partner-api-models/discussions/4427
